apps/src/templates/sectionProgress/sectionProgressRedux.js

```diff
@@ -0,0 +1,42 @@
+const SET_SECTION = 'sectionProgress/SET_SECTION';
+const SET_VALID_SCRIPTS = 'sectionProgress/SET_VALID_SCRIPTS';
+const SET_CURRENT_VIEW = 'sectionProgress/SET_CURRENT_VIEW';
+
+export const setSection = section => ({ type: SET_SECTION, section });
+export const setValidScripts = validScripts => ({ type: SET_VALID_SCRIPTS, validScripts });
+export const setCurrentView = viewType => ({ type: SET_CURRENT_VIEW, viewType });
+
+// Types of views of the progress tab
+export const ViewType = {
+  SUMMARY: "summary",
+  DETAIL: "detail",
+};
+
+const initialState = {
+  section: {},
+  validScripts: [],
+  currentView: ViewType.SUMMARY,
+};
```

Comment:
Food for thought - if you want to use something like Immutable.js for this module, it'd be good to set up now or in the near future before this gets very complex.  There's a learning curve to Immutable and I'm still trying to get a feel for when it's worth using, but probably worth taking a look at.

Reply:
I thought about it too. I actually think I should've used it for ManageStudentsTable, which has way more actions than this view will. Not convinced we need it here, but I'll keep it in mind before doing my next action. Wouldn't be hard to add soon since I already have test coverage.